commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java

```diff
@@ -179,6 +198,26 @@ public static ExecutorService newCachedThreadPoolWithMaxThreads(int maxThreads,
                 .build();
     }
 
+    /** Specialized cached executor which throws
+     * {@link java.util.concurrent.RejectedExecutionException} once max-threads have been exceeded.  This does not
+     * create a new span.
+     *
+     * If you have any doubt, this probably isn't what you're looking for. Best of luck, friend.
```

Comment:
😅

Reply:
I copied that from a different method.  I do love that warning :D